contracts/LoanCore.sol

```diff
@@ -714,22 +766,27 @@ contract LoanCore is
      *      Will validate loan state, perform accounting calculations, update storage and burn loan notes.
      *      Transfers should occur in the calling function.
      *
-     * @param loanId                The ID of the loan to repay.
-     * @param _amountFromPayer      The amount of tokens to be collected from the repayer.
-     * @param _amountToLender       The amount of tokens to be distributed to the lender (net after fees).
+     * @param loanId                 The ID of the loan to repay.
+     * @param _amountFromPayer       The amount of tokens to be collected from the repayer.
+     * @param _amountToLender        The amount of tokens to be distributed to the lender (net after fees).
+     * @param _interestAmount        The amount of interest to be paid.
+     * @param _paymentToPrincipal    The amount of principal to be paid.
      *
-     * @return data                 The loan data for the repay operation.
+     * @return data                  The loan data for the repay operation.
+     * @return loanRepaid            Bool denoting whether the loan is repaid.
      */
     function _handleRepay(
         uint256 loanId,
         uint256 _amountFromPayer,
-        uint256 _amountToLender
-    ) internal returns (LoanLibrary.LoanData memory data) {
+        uint256 _amountToLender,
+        uint256 _interestAmount,
+        uint256 _paymentToPrincipal
+    ) internal returns (LoanLibrary.LoanData memory data, bool loanRepaid) {
```

Comment:
What happens if `_interestAmount + _paymentToPrincipal != _amountFromPayer`? Related question, if we have the latter two values, what do we also need `_amountFromPayer` for?

Reply:
Yes correct, I made changes such that the `_amountFromPayer` is not needed and updated LC tests.